### PR TITLE
Implement Iterator support for Array

### PR DESCRIPTION
**Pull request by malept**
I implemented this a month ago but forgot to make a PR for it, after the ruby-sys PR was merged.

<del>Consider this a placeholder so that I remember to rebase this and so no one else duplicates my work :smile: </del>

### CODE REVIEW

**Comment by malept** [comment]
This is ready for review.

The Travis builds are failing due to a change unrelated to this.

**Comment by d-unseductable** [comment]
Wow looks amazing!

I will be able to test the code in several hours and then will let you know!

I think, I'll try to implement the same thing for `Hash` instead of having ruby functions like `each()`. It should make API much better for end-users.

File: src/class/array.rs [hunk 1]
```diff
@@ -160,3 +189,68 @@ impl Object for Array {
...
+/// let mut array = Array::new().push(Fixnum::new(1));
+/// array.push(Fixnum::new(2));
+/// array.push(Fixnum::new(3));
+/// let mut sum: i64 = 0;
+///
+/// for item in array.into_iter() {
+///     sum += item.to::<Fixnum>().to_i64();
+/// }
+///
+/// assert_eq!(sum, 6);
```
Comment (d-unseductable) [inline src/class/array.rs]: Missing final <code>```</code>

**Comment by d-unseductable** [comment]
Great! I will also add more examples to docs!

**Comment by d-unseductable** [comment]
@malept, please update to `0.7.8` if you need these changes. I've also pushed `FromIterator` for `Array` (@e8fdbe553d4f1bb708092a6c21e93de8fccf7436)

**Comment by malept** [comment]
> I've also pushed `FromIterator` for `Array`

Nice!

**Pull request merged**